keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/handlers/UnpinnedNotificationHandler.kt

```diff
@@ -2,21 +2,63 @@ package com.netflix.spinnaker.keel.slack.handlers
 
 import com.netflix.spinnaker.keel.slack.SlackUnpinnedNotification
 import com.netflix.spinnaker.keel.notifications.NotificationType
-import com.netflix.spinnaker.keel.slack.SlackNotifier
+import com.netflix.spinnaker.keel.slack.SlackService
+import com.slack.api.model.kotlin_extension.block.withBlocks
+import org.apache.logging.log4j.util.Strings
 import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Component
 
+/**
+ * Sends notification when unpinning an artifact
+ */
 @Component
 class UnpinnedNotificationHandler (
-  private val slackNotifier: SlackNotifier,
-  @Value("\${spinnaker.baseUrl}") private val spinnakerBaseUrl: String
-) : SlackNotificationHandler<SlackUnpinnedNotification> {
+  private val slackService: SlackService,
+  private val gitDataGenerator: GitDataGenerator,
+  @Value("\${spinnaker.baseUrl}") private val spinnakerBaseUrl: String,
+) : SlackNotificationHandler<SlackUnpinnedNotification>{
 
   override val type: NotificationType = NotificationType.UNPINNED_ARTIFACT
   private val log by lazy { LoggerFactory.getLogger(javaClass) }
 
   override fun sendMessage(notification: SlackUnpinnedNotification) {
-    TODO("Not yet implemented")
+    log.debug("Sending unpinned artifact notification for application ${notification.application}")
+
+    with(notification) {
+      val env = Strings.toRootUpperCase(targetEnvironment)
+      val artifactUrl = "$spinnakerBaseUrl/#/applications/${application}/environments/${latestArtifact?.reference}/${latestArtifact?.version}"
+      val username = slackService.getUserInfoByEmail(user)
+
+
+      val blocks = withBlocks {
+        header {
+          text("$env was unpinned", emoji = true)
+        }
+
+        section {
+          markdownText("*Version:* ~#${pinnedArtifact?.buildMetadata?.number}~ → <$artifactUrl|#${latestArtifact?.buildMetadata?.number}> " +
```

Comment:
Same comment on how this will render when any of these things are null. This question applies everywhere where you're using the null-safe operator `?.` in these notifications. I [tested this in the Kotlin playground](https://play.kotlinlang.org/#eyJ2ZXJzaW9uIjoiMS40LjAiLCJjb2RlIjoiXG5mdW4gbWFpbigpIHtcbiAgdmFsIHNvbWV0aGluZ051bGxhYmxlID0gbnVsbFxuICB2YWwgc3RyaW5nV2l0aFNhZmVOdWxsQ2hlY2sgPSBcIldoYXQgaGFwcGVucyBpZiAke3NvbWV0aGluZ051bGxhYmxlPy50b1N0cmluZygpfSBpcyBwYXNzZWQ/XCJcbiAgcHJpbnRsbihzdHJpbmdXaXRoU2FmZU51bGxDaGVjaylcbn0iLCJwbGF0Zm9ybSI6ImphdmEiLCJhcmdzIjoiIn0=) and it renders `null` in the middle of the string, which I'm not sure is what you want.